File: Developing-iOS-7-Apps-for-iPhone-and-iPad/imaginarium/imaginarium/ViewController.m
//
//  ViewController.m
//  imaginarium
//
//  Created by Tom Lai on 8/27/15.
//  Copyright (c) 2015 Lai. All rights reserved.
//

#import "ViewController.h"
#import "ImageViewController.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.destinationViewController isKindOfClass:[ImageViewController class]]) {
        ImageViewController *ivc = segue.destinationViewController;
        ivc.imageURL = [NSURL URLWithString:segue.identifier];
        ivc.title = segue.identifier;
    }
}


@end

File: animationObject/animationObject/ViewController.swift
//
//  ViewController.swift
//  animationObject
//
//  Created by Tom Lai on 10/20/15.
//  Copyright © 2015 Lai. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let transitionManager = TransitionManager()
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

    }
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return self.presentingViewController == nil ? UIStatusBarStyle.Default : UIStatusBarStyle.LightContent
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction override func unwindForSegue(unwindSegue: UIStoryboardSegue, towardsViewController subsequentVC: UIViewController) {
        transitionManager.presenting = false
    }

    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        let toViewController = segue.destinationViewController as UIViewController
        transitionManager.presenting = true
        toViewController.transitioningDelegate = self.transitionManager
    }
}
